program-runtime/src/executor_cache.rs

```diff
@@ -39,7 +39,7 @@ impl TransactionExecutorCache {
         }
     }
 
-    pub fn get(&self, key: &Pubkey) -> Option<Option<Arc<dyn Executor>>> {
+    pub fn get(&self, key: &Pubkey) -> Option<Option<Arc<LoadedProgram>>> {
```

Comment:
Not really necessary to be changed, as the transaction cache is going to replaced as well. But the second `Option` type is not needed anymore as we now have `LoadedProgramType::Invalid` as a proper tomb stone.